lib/Doctrine/DBAL/Schema/ForeignKeyConstraint.php

```diff
@@ -364,9 +364,9 @@ private function onEvent($event)
      */
     public function intersectsIndexColumns(Index $index)
     {
-        foreach ($index->getColumns() as $indexColumn) {
+        foreach ($index->getUnquotedColumns() as $indexColumn) {
             foreach ($this->_localColumnNames as $localColumn) {
-                if (strtolower($indexColumn) === strtolower($localColumn->getName())) {
+                if (strtolower($indexColumn) === strtolower($localColumn->trimQuotes($localColumn->getName()))) {
```

Comment:
I am not sure that I have to quote `$localColumn->getName()` this should always be unquoted?

Reply:
Can't really help with that due to my unfamiliarity with this code area. @deeky666 @morozov can you check this?

Reply:
I'm unfamiliar with this either but it feels bad. An unquoted column name is the value itself, a quoted column name is a SQL expression representing this value. Why would we want to mix them and compare by trimming quotes?

Reply:
@morozov maybe I did not understand you. I need them to ignore escaping because of #3387 while debugging I noticed that for up and down migrations it compares escaped or unescaped values and generates or not gerenates migration. This place checks if the column name is included in the foreign constraint and it should not matter if it is escaped or not should say it is in both cases. I just not sure about this `$localColumn->trimQuotes($localColumn->getName())`.

Also I am not sure this will not break other parts but tests looks green so maybe ok?

Reply:
I don't think it's a valid case for DBAL. Right now there's no difference between quoted and unquoted identifiers in the DBAL type system but it doesn't mean it should handle them transparently. Most likely, there's a lot of other cases where mixing quoted and unquoted identifiers will cause unexpected behavior.

Instead of changing the comparison logic in DBAL, I'd try to fix the inconsistency on the calling side.

Reply:
Isn't this a valid case where a database allows reserved words when they are quoted?